src/reference/modules/administration/pages/security.adoc

```diff
@@ -1,178 +1,126 @@
-
-=== Provisioning an agent with pre-established trust
-
-This section expects that the server-side setup has been done. At allows going a bit further by providing a hash of the server key at installation
-to validate the server identity from the first connection, and avoid trusting the
-first host it connects to.
-
-It requires providing local changes to the agent as part of the provisioning process.
-Here policy server's refers to you node's server, it can be a relay or a root server.
-
-Compute policy server's hash key with:
+* To check if the key pinning is correctly configured on the node, you can check the output of `rudder agent info` (it should indicate **full**):
 
 ----
+[root@rudder ~]# rudder agent info
 
-cf-key -p /var/rudder/cfengine-community/ppkeys/localhost.pub
-
+[...]
+        Key pinning: full
+[...]
 ----
 
-Copy `/var/rudder/cfengine-community/ppkeys/localhost.pub` from the policy server server on each provisioned node to `/var/rudder/cfengine-community/ppkeys/root-${HASH}.pub`
-
-So that it looks like:
+Now you can set your policy server with `rudder agent policy-server mypolicyserver`
+and the agent will only accept the connection if it matches the provided key hash, on both communication protocols.
 
-----
+==== Pre-provision a node on the server
 
-# cat /var/rudder/cfengine-community/ppkeys/root-MD5\=1ec2213e08921bd3444861f7b4a60919.pub
------BEGIN RSA PUBLIC KEY-----
-MIICCgKCAgEA78g1gmG98Sh4hso8mYGagj98M+SZU7mklbC5Ylv90mecsLD9QlcZ
-j64z5uABclz5lErdbtVu7ix4Tk5PyrTW0vbh40tqa48oifPl4iA7hQUmYpnnO8Sp
-h/HCcHGsiLrW5PytDN2JcOaRZz03cOaoze29KkQgjav+DNZdzqV79aVnujaRqTPY
-G6B3pInhVeKgiuJkXQEiql4f4GdBHZTkEDz2ammjqu507NL8iPkJs1mE8N0Q0CtC
-mgXNqzS6bIzTLMgBU1MK77NLsPs2GsNo9x6V/CuutmUHwlccdf1NVyhO9EpzxY1C
-TFzojgPT62pKR5Ehgl9bHyVGF1RY0kxwKfXJqyVZg87z7jh3YjyzG4RWsihNyiRe
-HUuKK+rpMj+QwRbbt/97siDxHb6OShKDEWJoR8aj//vY03c79zXtZtoAcpPDp5Et
-jNUEYqnVZabEuryaUsAOD0Dyx3twWEm1DW+4PJYUYDlNtcuD8X3kvyOnjBy9AbbR
-irG1iZFVLzyNaX/0ijdY94+iw2c1Ga4HQp6MAdkKiaJlN7pkRksd4778dZRimYiu
-dD42Nb36YQca50d0sHKalWFTBaT+ksRSrFZVs1Wb8yE9KkJf4bYLRYLhIdYTra22
-C9o5xHCOTDecATXMg0gGQHbjm0x0a1nt+X1gyyjNfHLX13n5as9JXf0CAwEAAQ==
------END RSA PUBLIC KEY-----
-
-----
+To automate node acceptation on the server while checking
+the node's identity, you can use the https://docs.rudder.io/api/#tag/Nodes/operation/createNodes[node creation] API.
+You can make a call to this API in your node provisioning process, and provide
+the `agentKey` parameter (which should contain the content of `/opt/rudder/etc/ssl/agent.cert` on the node), and the `accepted` status.
 
-Now you can set your policy server with `rudder agent policy-server mypolicyserver`
-and the agent will only accept the connection if it matches the provided key hash.
+This way, once the node sends its first inventory, if it matched the node id and certificate, it will be automatically accepted. Note that this creation call also allows pre-setting node properties.
```

Comment:
this is misleading - the node is already existing within Rudder

This way, once the node sends its first inventory, if the node id and certificate match the pre-provisionned entry, the node inventory will be update.
Please note that you can also pre-defined node properties using this API, to ensure that the pre-provisionned node is already in the correct groups